docs/cudf/source/developer_guide/testing.md

```diff
@@ -291,3 +291,12 @@ def test_bug_in_current_and_maybe_future_versions(...):
 ```
 
 If pandas makes a bugfix release and fixes this, then we'll see it in CI immediately, patch it, and bump `PANDAS_CURRENT_SUPPORTED_VERSION` which also usually happens during pandas upgrades.
+
+## Test Parallelism: Why cuDF Tests Use `-n 8`
+When running cuDF tests with `pytest-xdist`, the number of parallel workers is explicitly set to `-n 8` rather than using `-n auto`. While `-n auto` launches one worker per available CPU core, cuDF tests are generally constrained by the amount of available GPU memory per worker.
```

Comment:
If we feel this strongly about documenting, perhaps we could go ahead and compute the desired number of workers from the available GPU memory? We can query nvidia-smi and use that information to pick a number of workers.

Generally I want to avoid putting too many references to specific hardware in our docs, as our CI cluster will change over time.

Reply:
Proposal:

```bash
gpu_mem_mib=$(nvidia-smi --query-gpu=memory.total --format=csv,noheader,nounits | head -n1)
gpu_mem_gb=$((gpu_mem_mib / 1024))
mem_gb_per_worker=3
num_workers=$((gpu_mem_gb / mem_gb_per_worker))
```

Reply:
Thanks for the proposal, I'll give it a shot

Reply:
@Matt711 is your plan to try implementing the automated calculation in this PR instead of doing it in a follow-up?

Reply:
I meant to do this in this PR, but didn’t get to it. I think the docs in this PR look good, so let’s handle the automated calculation implementation in a follow-up. I opened #19676 to track. @bdice WDYT?

Reply:
I agree, I'd rather just implement it and skip docs altogether.